core/rpc/client/gateway/http/client.go

```diff
@@ -120,20 +120,16 @@ func (g *GatewayHttpClient) GetAuthParameter(ctx context.Context) (*types.Gatewa
 		return nil, err
 	}
 
-	if r.Error != "" {
-		return nil, errors.New(r.Error)
-	}
-
 	return r.Result, nil
 }
 
-// gatewayAuthPostResponse defines the response of POST request for /auth
-type gatewayAuthPostResponse struct {
-	Error string `json:"error"`
+// gatewayErrResponse defines the response of gateway error.
+type gatewayErrResponse struct {
+	// to be compatible with google.golang.org/genproto/googleapis/rpc/status.Status
+	Message string `json:"message"`
```

Comment:
Confirmed this matches kgw `ResponseError`: https://github.com/kwilteam/kgw/pull/29/files#diff-7ed3dcaadf0b52ddf9731d830f75cdec87978777990f6a397d5a2beb6ad6975bR27-R29

and kgw marshals that type for errors in https://github.com/kwilteam/kgw/pull/29/files#diff-8b7f25bac2ae04c2ce2abb628c6aa28103fd113103a62f3e9d232fa995684915R67

:heavy_check_mark: